content/1_docs/2_cookbook/9_setup/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,468 @@
+
+Static properties and methods can be called directly without creating an instance of a class with the `new` keyword.
+
+Examples for static properties in the `Page` class are `$methods`  and `$models`, examples for static methods the `factory()` or the `model()` functions.
+
+The syntax for calling static properties and methods is different from non-static ones:
+
+```php
+$p = Page::factory([
+  'slug' => 'projects',
+  'content' => [
+      'title' => 'Projects'
+  ]
+]);
+dump($p);
+```
+
+Instead of the arrow syntax used for methods that are called on an instance of a class like `$page->update()`, a double colon (or (link: https://en.wikipedia.org/wiki/Scope_resolution_operator text: "Paamayim Nekudotayim")) is used to access a static method or property of a class.
+
+<info>
+Kirby's Toolkit comes with multiple classes that have only static methods, for example the `Str` or the `Html` classes.
+</info>
+
+## A closer look at the methods and properties of the `Page` class
+
+If you have a closer look at the properties defined in the `Page.php` file, you will notice that apart from the two static ones mentioned above which are public, all others are protected. If you then inspect the methods in this file, you will find that there is a public method for all these properties, the getters we already explained in the context of the `Book` class above.
+
+For example, we can access the `$parent` property with `$page->parent()`, the `$template` property with `$page->template()` and the `$inventory` property with `$page->inventory()` etc.
+
+But what about setting properties like in our `Book` example above where we could call `$book->setPrice($price)` to assign a new price to the object at runtime? Can we change a `Page` property by calling `$page->setTitle()`. Spoiler: No, we can't.
+
+All setter methods, like `setProperties()`, `setSlug()` etc. are protected functions that cannot be used outside the class and its child classes. The object is therefore immutable.
+
+The only way to change a Page's properties is thus to actually modify the page on disk and then store the result of the modification in a (new) variable.
+
+### Objects are immutable
+
+Try this in a template in a Starterkit:
+
+```php
+$kirby->impersonate('kirby');
+$p = page('notes');
+dump($p->title());
+$p->update([
+    'title' => 'Blog'
+]);
+dump($p->title());
+```
+You will notice that nothing will have changed, although the title was actually changed on disk (check the content text file to verify). Only after you reload the page, the changes will be reflected. That is because a new object with the new properties will be created at runtime of the script.
+
+To fix this, we can do it like this instead:
+
+```php
+$p = page('notes');
+dump($p->title());
+$p = $p->update([
+    'title' => 'Blog'
+]);
+dump($p->title());
+```
+Now the changes will be reflected without reloading the page, because we override the original `$p` variable and assign the new page object to it.
```

Comment:
```suggestion
Now the changes will be reflected without reloading the page, because we overwrite the original `$p` variable and assign the new page object to it.
```